src/frontend/pages/blogs/edit-a-blog/EditBlog.js

```diff
@@ -0,0 +1,317 @@
+            <ImageCard>
+              <ImagePreview
+                src={URL.createObjectURL(uploadedImageURL)}
+                onDelete={() => {
+                  imageDelete(); 
+                }}
+              />
+          </ImageCard>
+          }
+        </div>
+        ]; 
+      }, [currentImageURL, uploadedImageURL]);
+
+
+    const errorList = useMemo(() => {
+      return [
+        ...errors.map((error, index) => {
+          return (
+            <ErrorMessage key={error}>
+              {`${index+1}. `}{error}
+            </ErrorMessage>
+          );
+        })
+      ];
+    }, [errors]);
+
+  const updateVisibilityCatagory = (value) => {
+    switch (value){
+      case 2:
+        setVisibility("Hidden")
+        break;
+      default:
+        setVisibility("Public")
+        break;
+    }
+  }
+
+  const updateClosedCatagory = (value) => {
+    switch (value){
+      case 1:
+        setClosed(true)
+        break;
+      default:
+        setClosed(false)
+        break;
+    }
+  }
+
+  return (
+    <EditBlogPage>
+      <TopRow>
+        <Button cancel onClick={closeForm}>
+          &#60; Back
+        </Button>
+      </TopRow>
+
+      <EditBlogBody>
+        <FormContainer title="Blog Title (required)">
+          <TextInput
+            value={blogTitle}
+            onChange={setblogTitle}
+            placeholder="Blog Title"
+          />
+        </FormContainer>
+
+        <FormContainer title="Author (required)">
+          <TextInput
+            value={author}
+            onChange={setAuthor}
+            placeholder="Author"
+          />
+        </FormContainer>
+
+        <FormContainer title="Summary (required)">
+          <BlogInfo>
+            <TextInput
+            value={summary}
+            onChange={setSummary}
+            placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit..."
+            multiLine={true}
+          />
+          </BlogInfo>
+        </FormContainer>
+
+        <FormContainer title="Date Published (required): Must be day-month-year">
+          <TextInput
+            value={date}
+            onChange={setDate}
+            placeholder="23-May-2021"
+          />
+        </FormContainer>
+
+        <FormContainer title="Link (required)">
+          <TextInput
+            value={link}
+            onChange={setLink}
+            placeholder="https://..."
+          />
+        </FormContainer>
+
+        <FormContainer title="Closed (required: open by default)">
+          <Selector
+                value={closed? 1 : 2}
+                items={[{id: 1, text: "True"}, {id: 2, text: "False"}]}
+                onSelect={updateClosedCatagory}
+              />
+        </FormContainer>
+
+        <FormContainer title="Visibility (required)">
+          <Selector
+                value={visibility === "Hidden"? 1 : 2}
+                items={[{id: 1, text: "Hidden"}, {id: 2, text: "Public"}]}
+                onSelect={updateVisibilityCatagory}
```

Comment:
This definitely works. 

It is just somewhat prone to mistakes/error. It's better if we associated "hidden" with 1, and "public" with 2 using some common data structure.

For a later feature, you can do this:
```
const VISIBILITY = {
    hidden: {id: 1, text: "Hidden"},
    public: {id: 2, text: "Public"}
}
```

and have:

```
import VISIBILITY from './consts.js'

<Selector
                value={visibility} // The ID of the current visibility status
                items={Object.values(VISIBILITY)}
/>
```

This way, {"hidden", 1} {"public", 2} values will always be associated with each other, and it's much harder for any new developer working on this feature to mess it up.

It's also better since  `VISIBILITY` is defined in 1 place to avoid mistakes/errors. Just imagine if this file had 3 different:

```
<Selector
                value={visibility === "Hidden"? 1 : 2} 
/>
<Selector
                value={visibility === "Hidden"? 1 : 2} 
/>
<Selector
                value={visibility === "Hidden"? 1 : 2} 
/>
```

This is error-prone, since modifying any one `visibility === "Hidden"? 1 : 2` by accident suddenly leaves the logic inconsistent.

Again, no need to change for this PR.

Reply:
Yeah, if I had to choose my least favourite part of the PR it would be this. The <Selector> was just hard to use for this use case and that was the best I could come up with at the time